pytest-embedded-idf/pytest_embedded_idf/serial.py

```diff
@@ -266,3 +266,29 @@ def is_target_flashed_same_elf(self) -> bool:
             elf_sha256.update(fr.read())
 
         return flash_elf_sha256 == elf_sha256.digest()
+
+
+class LinuxSerial:
```

Comment:
Here I changed my mind a bit. I think make this class inherit `DuplicateStdoutPopen` would be better.  `DuplicateStdoutPopen` has a few methods that may used by others. we need to override them. For example:

1. `write()`. when calling `dut.write()` it would actually call the dut.serial.write, in linux target, we don't support write yet. We could raise a `ValueError` instead.
2. `close` and `terminate` are used for terminating. For your current solution, there's no way to terminate the process, which would hang for a long time.

Reply:
> in linux target, we don't support write yet. We could raise a ValueError instead.

wrong. We do support write now. https://github.com/espressif/esp-idf/blob/master/components/esp_system/host_test/test_esp_system/main/esp_system_test.c

Reply:
In this case we have to define a new `LinuxDut`, right ?

Reply:
In the current approach I use Dut instance, not SerialDut

Reply:
In `Dut.write` you are actually writing to the message queue. Not the popen process itself

Reply:
Use `Dut` is fine